service/dap,rpc1,rpc2: refactor creating RPC clients

alexandear opened the pull request:
The PR refactors RPC code for creating a Client:

- Rename `rpc2.NewClient` to `rpc2.NewTestClient` as it's being used only in tests.
- Rename `rpc1.NewClient` to `rpc1.NewTestClient` as it's being used only in tests.
- Move `dap.NewClient` to `daptest.NewClient`.
- Rename `rpc1/readme.txt` to `rpc1/doc.go`
- Use `t.Fatal` instead of `log.Fatal`

aarzilli reviewed (commented):
We're only using it in tests, it seems the only effect of this is increasing the number of lines of code. Maybe we should move them to a _export_test.go file, so we don't accidentally call them in real code and leave them as they are.

alexandear commented:
> We're only using it in tests, it seems the only effect of this is increasing the number of lines of code. Maybe we should move them to a _export_test.go file, so we don't accidentally call them in real code and leave them as they are.

Refactored, please re-review. It's impossible to move to `_export_test.go` because Go doesn't export functions from a `_test.go` file. Therefore, I moved `NewClient` to a separate package intended for use only in tests.

aarzilli commented on service/rpc1/client.go:
```diff
@@ -23,10 +22,10 @@ type RPCClient struct {
 
 // NewClient creates a new RPCClient.
-func NewClient(addr string) *RPCClient {
+func NewClient(t testing.TB, addr string) *RPCClient {
```
Why not do the same thing to service/rpc2.NewClient as well? They could be renamed to NewTestClient.

alexandear replied:
Agree, thanks.
Renamed.

aarzilli reviewed (approved):
LGTM

derekparker commented:
This package is public and versioned and this symbol is exported, removing this symbol because we're not using it does not guarantee we won't break other programs who may be using this to initialize a client. Overall I'm against removing this right now, and I'm not a huge fan of exporting a function in this package that is only used in tests. If that's the case, just implement a test helper directly.

derekparker reviewed (changes requested):
See above comment.

alexandear commented:
@derekparker Agree.

I found that `rpc1.NewClient` and `rpc2.NewClient` are used:

- https://sourcegraph.com/search?q=context:global+%22rpc1.NewClient%22+lang:go&patternType=standard&sm=1&groupBy=repo
- https://sourcegraph.com/search?q=context:global+%22rpc2.NewClient%22+lang:go&patternType=standard&sm=1&groupBy=repo

So, closing this PR as it contains breaking changes.
